code/code/spec/spec_objs_thief_quest.cc

```diff
@@ -5,73 +5,63 @@
 #include "extern.h"
 #include "obj_base_weapon.h"
 
-int thiefQuestWeapon(TBeing *victim, cmdTypeT command, const char *arg, TObj *object, TObj *) {
-  wearSlotT limb = WEAR_NOWHERE;
-
-  if (command != CMD_STAB && command != CMD_BACKSTAB && command != CMD_SLIT)
+int thiefQuestWeapon(TBeing *victim, cmdTypeT command, const char *arg, TObj *object, TObj *thief) {
+  // Thief will only == nullptr when proc is called on the generic command proc
+  // check. Don't want proc to execute in that instance - should only happen in
+  // response to skill use. Otherwise it'll execute twice as often as it should.
+  if ((command != CMD_STAB && command != CMD_BACKSTAB && command != CMD_SLIT) || !thief || !victim || !object)
     return false;
 
-  if (command == CMD_STAB)
-    limb = static_cast<wearSlotT>(reinterpret_cast<uintptr_t>(arg));
+  auto *weapon = dynamic_cast<TBaseWeapon *>(object);
+  auto *wielder = weapon ? dynamic_cast<TBeing *>(weapon->equippedBy) : nullptr;
```

Comment:
I wonder if dynamic_casting a nullptr is UB?

Reply:
Appears it's defined and acceptable:
>A null pointer value is converted to the null pointer value of the destination type by dynamic_cast.

https://docs.microsoft.com/en-us/cpp/cpp/dynamic-cast-operator

Reply:
In that case, you could write `dynamic_cast<TBeing*>(weapon && weapon->equippedBy)` - though please don't, it'd look abominable.

Reply:
This line was removed in https://github.com/sneezymud/sneezymud/pull/398/commits/950e22304888a94fcc672a6f3418adf574d64baa